streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamsPartitionAssignor.java

```diff
@@ -573,14 +586,24 @@ private ApplicationState buildApplicationState(final TopologyMetadata topologyMe
         ));
 
         return new DefaultApplicationState(
-            assignmentConfigs.toPublicAssignmentConfigs(),
+            publicAssignmentConfigs,
             logicalTasks,
             clientMetadataMap
         );
     }
 
-    private static void processStreamsPartitionAssignment(final Map<UUID, ClientMetadata> clientMetadataMap,
-                                                          final TaskAssignment taskAssignment) {
+    private void processStreamsPartitionAssignment(final org.apache.kafka.streams.processor.assignment.TaskAssignor assignor,
+                                                   final TaskAssignment taskAssignment,
+                                                   final AssignmentError assignmentError,
+                                                   final Map<UUID, ClientMetadata> clientMetadataMap,
+                                                   final GroupSubscription groupSubscription) {
+        if (assignmentError == AssignmentError.UNKNOWN_PROCESS_ID || assignmentError == AssignmentError.UNKNOWN_TASK_ID) {
+            assignor.onAssignmentComputed(new GroupAssignment(Collections.emptyMap()), groupSubscription, assignmentError);
+            log.error("Task assignment returning empty GroupAssignment and failing due to error {}", assignmentError);
```

Comment:
Technically speaking we're not returning any GroupAssignment at all for the "task assignment" ie the rebalance, since we throw this exception instead of returning anything from #assign. A subtle difference in wording but I think it's worth clarifying to avoid confusion here

```suggestion
            log.error("Rebalance failed due to task assignor returning assignment with error {}, assignor callback will receive empty GroupAssignment due to this error", assignmentError);
```